DevAcademy/Scenes/PlaceDetail/PlaceDetailViewModel.swift

```diff
@@ -2,24 +2,161 @@ import SwiftUI
 
 struct PlaceDetailViewModel: DynamicProperty {
     
-    @EnvironmentObject private var placesObject: PlacesObservableObject
+    // MARK: PROPERTIES
+    @EnvironmentObject private var placesObservableObject: PlacesObservableObject
     @State var isTappedFavorite = false
     
     var place: Place
     init(place: Place) {
         self.place = place
     }
     
+    
+    // MARK: PLACE MAIN INFORMATION
+    // IMAGE
+    var placeImage: String {
+        if imageIsFetch {
+            return place.attributes.imageURL!
+        }
+        return "Obrázek není k dispozici"
+    }
+    
+    var imageIsFetch: Bool {
+        if attributeIsFetch(place.attributes.imageURL) {
+            return true
+        }
+        return false
+    }
+    
+    // TITLE
     var placeName: String {
-        place.properties.nazev
+        place.attributes.title
+    }
+    
+    // KIND
+    var kindOfPlace: String {
+        place.attributes.kind.rawValue
+    }
+    
+    // STREET
+    var placeStreet: String {
+        if attributeIsFetch(place.attributes.street) {
```

Comment:
namisto toho force unwrappu muzes pouzit klasicke unwrappovani (if let):
```Swift
var placeStreet: String {
    if let street = place.attributes.street {
        return street
    } else {
        return ""
    }
}
```
a nebo to pripadne jeste zkratit pomoci operatoru `??`
```Swift
var placeStreet: String {
    place.attributes.street ?? ""
}
```